### DISCUSSION TITLE
Handle twig extension

### DISCUSSION THREAD

**Discussion by valeriosillari** [megahertz/twigjs-loader#15] [Q&A]
Hi, I hope this is not off-topic.

I've started using twigjs-loader and all seems fine.
I can call twig files via webpack.

But some twig files got an error.
These twig files are using a custom _twig extension_ I've created.
Webpack and the loader seems not to know where this extension is (meaning path and how to treat it) and return an error.

Question:
Can this package handle custom twig extensions? Is something that this package should take care of?

Please any tips and support is very welcome

Best

**Comment by megahertz** [comment]
Do you mean template file extension or twig.js extension? If you mean twig.js extension, try to load this extension in webpack config, I think it should work fine if it's registered before webpack starts transpilling.

**Reply by valeriosillari** [reply]
Good question, probably my question was not so clear ...

I mean "twig extensions"
https://symfony.com/doc/current/templating/twig_extension.html

I got one used in my application to set some custom date format (default symfony + twig logic).
Works perfectly when used in a twig file by symfony.

When the same file is called by twig loader, I got an error from webpack (webpack does not know where to find and use the extension).

I will check the webpack logic as suggested ("register before webpack starts transpilling").
Just wondering if this an issue related to twig loader package or more generally about webpack config :)

**Reply by megahertz** [reply]
So, you should implement a twig.js extension first

**Comment by sal-lochbaum** [comment]
For anyone else stumbling over discussions without solutions ;-)

Do something like this:

```
                {
                    test: /\.twig$/,
                    use: {
                        loader: 'twigjs-loader',
                        options: {
                            /**
                             * @param {object}  twigData        Data passed to the Twig.twig function
                             * @param {string}  twigData.id     Template id (relative path)
                             * @param {object}  twigData.tokens Parsed AST of a template
                             * @param {string}  dependencies    Code which requires related templates
                             * @param {boolean} isHot           Is Hot Module Replacement enabled
                             * @return {string}
                             */
                            renderTemplate(twigData, dependencies, isHot) {
                                return `
                                    ${dependencies}
                                    const Twig = require('twig');

                                    Twig.extendFunction("custom_function", function(a,b,c,d) {
                                        return 'some return value';
                                    });

                                    var tpl = Twig.twig(${JSON.stringify(twigData)});
                                    module.exports = function(context) { return tpl.render(context); };
                               `;
                            },
                        },
                    }
                }
```
